pandas/core/series.py

```diff
@@ -1625,6 +1625,13 @@ def dot(self, other):
         else:  # pragma: no cover
             raise TypeError('unsupported type: %s' % type(other))
 
+    def __matmul__(self, other):
+        from pandas.core.frame import DataFrame
+        dot_types = (Index, DataFrame, Series, np.ndarray, list)
```

Comment:
Your tests need to ensure that for every supported dtype, this operation works.  I don't think that's the case currently.